public/app/percona/ui-events/EventStore.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+const cache = new Map<string, any>();
+
+export enum EventType {
+  NotificationError = 'NOTIFICATION_ERROR',
+  Fetching = 'FETCHING',
+  DashboardUsage = 'DASHBOARD_USAGE',
+}
+
+export const EventStore = {
+  add: (event: EventType, payload: any) => {
+    let current: any[] = cache.get(event);
+    if (!current) {
+      current = [];
+      cache.set(event, current);
+    }
+
+    current.push(payload);
```

Comment:
I know by pushing to the array it will end up in the map due to the reference being the same, but maybe let's be explicit:

```
if (!current) {
      current = [];
}

cache.set(event, current);
cache.set(event, current);
```

Reply:
I think it is not necessary to add `cache.set(event, current);` if see this code I would have more questions.